src/main/java/de/mari_023/fabric/ae2wtlib/util/ContainerHelper.java

```diff
@@ -135,15 +135,10 @@ private I getHostFromPlayerInventory(PlayerEntity player, ContainerLocator locat
         }
 
         String currentTerminal = WUTHandler.getCurrentTerminal(it);
-        //TODO do something generic, I don't want to hardcode everything
-        if(interfaceClass.isAssignableFrom(WCTGuiObject.class) && currentTerminal.equals("crafting"))
-            return interfaceClass.cast(new WCTGuiObject((ItemWT) it.getItem(), it, player, locator.getItemIndex()));
 
-        if(interfaceClass.isAssignableFrom(WPTGuiObject.class) && currentTerminal.equals("pattern"))
-            return interfaceClass.cast(new WPTGuiObject((ItemWT) it.getItem(), it, player, locator.getItemIndex()));
+        if(interfaceClass.isAssignableFrom(WCTGuiObject.class) && (currentTerminal.equals("crafting") || currentTerminal.equals("pattern") || currentTerminal.equals("interface")))
+            return interfaceClass.cast(new WCTGuiObject((ItemWT) it.getItem(), it, player, locator.getItemIndex()));
 
-        if(interfaceClass.isAssignableFrom(WITGuiObject.class) && currentTerminal.equals("interface"))
-            return interfaceClass.cast(new WITGuiObject((ItemWT) it.getItem(), it, player, locator.getItemIndex()));
```

Comment:
This does not work. it has to create different objects for different Terminals

Reply:
we had this issue before, but I think I should comment my code better

Reply:
+